keras_cv/models/segmentation/deeplab_v3_plus.py

```diff
@@ -0,0 +1,267 @@
+    scaled_masks = scaled_masks[
+        :,
+        offset[0]:offset[0] + output_size[0],
+        offset[1]:offset[1] + output_size[1],
+        :]
+
+    output_masks = tf.image.pad_to_bounding_box(
+        scaled_masks, 0, 0, output_size[0], output_size[1])
+    # Remove padding.
+    output_masks = output_masks[1::]
+    return output_masks
+
+def resize_fn(image, cls_seg):
+    image, image_info = resize_and_crop_image(
+        image, image_size[:2], image_size[:2], 0.5, 2.0
+    )
+    image_scale = image_info[2, :]
+    offset = image_info[3, :]
+    cls_seg += 1
+    cls_seg = cls_seg[tf.newaxis, ...]
+    cls_seg = resize_and_crop_masks(cls_seg, image_scale, image_size[:2], offset)
+    cls_seg = cls_seg[0]
+    cls_seg -= 1
+    cls_seg = tf.where(
+        tf.equal(cls_seg, -1), 255 * tf.ones_like(cls_seg), cls_seg
+    )
+    return image, cls_seg
+
+def flip_fn(image, cls_seg):
+    if tf.random.uniform([], minval=0, maxval=1, dtype=tf.float32) > 0.5:
+        image = tf.image.flip_left_right(image)
+        cls_seg = tf.reverse(cls_seg, axis=[1])
+    return image, cls_seg
+
+def proc_train_fn(examples):
+    image = examples.pop("image")
+    image = tf.cast(image, tf.float32)
+    # image = resize_layer(image)
+    cls_seg = examples.pop("class_segmentation")
+    cls_seg = tf.cast(cls_seg, tf.float32)
+    # cls_seg = resize_layer(cls_seg)
+    image, cls_seg = resize_fn(image, cls_seg)
+    image, cls_seg = flip_fn(image, cls_seg)
+    cls_seg = tf.cast(cls_seg, tf.uint8)
+    sample_weight = tf.equal(cls_seg, 255)
+    zeros = tf.zeros_like(cls_seg)
+    cls_seg = tf.where(sample_weight, zeros, cls_seg)
+    return image, cls_seg
+
+def proc_eval_fn(examples):
+    image = examples.pop("image")
+    image = tf.cast(image, tf.float32)
+    image = resize_layer(image)
+    cls_seg = examples.pop("class_segmentation")
+    cls_seg = tf.cast(cls_seg, tf.float32)
+    cls_seg = resize_layer(cls_seg)
+    cls_seg = tf.cast(cls_seg, tf.uint8)
+    sample_weight = tf.equal(cls_seg, 255)
+    zeros = tf.zeros_like(cls_seg)
+    cls_seg = tf.where(sample_weight, zeros, cls_seg)
+    return image, cls_seg
+
+train_ds = train_ds.map(proc_train_fn, num_parallel_calls=tf.data.AUTOTUNE)
+train_ds = train_ds.batch(global_batch)
+
+eval_ds = eval_ds.map(proc_eval_fn, num_parallel_calls=tf.data.AUTOTUNE)
+eval_ds = eval_ds.batch(global_batch)
+
+train_ds = train_ds.shuffle(8)
+train_ds = train_ds.prefetch(2)
+
+with strategy.scope():
+    lr_decay = tf.keras.optimizers.schedules.PiecewiseConstantDecay(
+        boundaries=[30000 * 16 / global_batch],
+        values=[base_lr, 0.1 * base_lr],
+    )
+    # backbone = keras_cv.models.ResNet50V2(
+    #     include_rescaling=True, weights="imagenet", include_top=False,
+    #     input_shape=[512, 512, 3]
+    # ).as_backbone()
+    # model = DeepLabV3(classes=21, backbone=backbone, include_rescaling=True)
+    model = DeeplabV3Plus(image_size=512, num_classes=21)
```

Comment:
Looks like the backbone is missing?

Reply:
@DavidLandup0 Updated

Reply:
I actually updated with a new training script as well, so now you should be able to reproduce it

Reply:
Awesome, thanks! I did notice a difference in some longer runs, where mine used to plateau faster than the TB logs, and was wondering what was going on...
Can't wait to update the `basic_training.py` to include these as KPLs!

Reply:
KPLs is a great feature, makes things much easier!
I do notice that segmentation mostly just use simple resizing, so updating training to only use KPL is good to go. 
However we're rolling back KPL for detection models since it has been consistently trained with lower precision, and we need to figure out the logic difference.

Reply:
I've also noticed that when migrating the sem seg script fully (initial PR that was dropped), the augmentations seem to have slightly reduced the performance of the model. Visualizing the images and maps seemed to have verified a perfectly fine operation for each of them, but there might be some side effect that we aren't considering.

The runs with only flipping practically always outperformed the rest, which is kind of surprising...